src/models/departments/reducer.js

```diff
@@ -0,0 +1,39 @@
+import { fromJS } from 'immutable';
+import {
+  FETCH_DEPARTMENTS_SUCCESS,
+  FETCH_DEPARTMENTS,
+  FETCH_DEPARTMENTS_ERROR,
+} from './constants';
+
+export const initialState = fromJS({
+  count: undefined,
+  error: false,
+  errorMessage: '',
+  list: [],
+  loading: false,
+});
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_DEPARTMENTS:
+      return initialState.set('loading', true);
+
+    case FETCH_DEPARTMENTS_SUCCESS:
+      return state
+        .set('count', action.payload.count)
+        .set('error', false)
+        .set('errorMessage', '')
+        .set('list', fromJS(action.payload.results))
+        .set('loading', false);
+
+    case FETCH_DEPARTMENTS_ERROR:
+      return state
+        .set('count', undefined)
```

Comment:
Why update `count` if we don't update `list`? Those 2 will get out of sync on error.

Reply:
True. `count` Doesn't have to be updated.